valohai_yaml/objs/utils.py

```diff
@@ -15,20 +16,31 @@ def consume_array_of(source: SerializedDict, key: str, type: Type[TItem]) -> Lis
     return [type.parse(datum) for datum in source.pop(key, ())]
 
 
-def check_type_and_listify(source: Optional[Iterable[Any]], type: Type[T]) -> List[T]:
-    """Check that all items in the `source` iterable are of the type `type`, return a list."""
+def check_type_and_listify(
+    source: Optional[Iterable[Any]],
+    type: Type[T],
+    parse: Optional[Callable[[Any], T]] = None,
+) -> List[T]:
+    """
+    Check that all items in the `source` iterable are of the type `type`, return a list.
+
+    If `parse` is given, and the item is not of the type, that function is called to parse it to one.
+
+    """
     if source is None:
         return []
     out = []
     for item in source:
         if not isinstance(item, type):
-            raise TypeError(f"{item} not a {type}")
+            if not parse:
+                raise TypeError(f"{item} not a {type}")
+            item = parse(item)
```

Comment:
should we check that the resulting item is of that `type` or you think that is a overkill?

I mean, at least I assumed that would be the case before I read the implementation

Reply:
I was thinking about doing it, and decided it might be somewhat overkill.

Since you were thinking about it too, there's now a type assertion too ✨